File: Problems 2000-2099/2016. Maximum Difference Between Increasing Elements/solution.ts
function maximumDifference(nums: number[]): number {
    let minSoFar = Number.MAX_SAFE_INTEGER;
    let res = -1;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > minSoFar) {
            res = Math.max(res, nums[i] - minSoFar);
        }
        minSoFar = Math.min(minSoFar, nums[i]);
    }
    return res;
};

File: Problems 2000-2099/2016. Maximum Difference Between Increasing Elements/solution.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        n = len(nums)
        if n < 2:
            return -1
        
        minSoFar = nums[0]
        res = -1
        for num in nums:
            if num > minSoFar:
                res = max(res, num - minSoFar)
            minSoFar = min(minSoFar, num)
        
        return res